.semaphore/semaphore.yml

```diff
@@ -0,0 +1,849 @@
+            - cd ws
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test Readers
+          commands:
+            - cd readers
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+
+  - name: Messaging Changes
+    dependencies: []
+    run:
+      when: "change_in('pkg/messaging/message.proto', {pipeline_file: 'ignore'})"
+    task:
+      jobs:
+        - name: Test Messaging
+          commands:
+            - cd pkg/messaging
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test CoAP
+          commands:
+            - cd coap
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test MQTT
+          commands:
+            - cd mqtt
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+        - name: Test WS
+          commands:
+            - cd ws
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov            
+        - name: Test OPC-UA
+          commands:
+            - cd opcua
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov            
+        - name: Test Twins
+          commands:
+            - cd twins
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov            
+        - name: Test Consumers
+          commands:
+            - cd consumers
+            - go test ./... --race -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html
+            - cache restore codecov
+            - ./codecov
+
+  - name: Push Docker Images
+    dependencies:
+      - Setup
+    run:
+      when: "branch = 'master'"
+    task:
+      jobs:
+        - name: Push Docker Images
+          commands:
+            - make -j$NPROC latest
+
+  - name: Compile Check For RabbitMQ
+    dependencies:
+      - Setup
+    task:
+      jobs:
+        - name: Compile For RabbitMQ
+          commands:
+            - MF_BROKER_TYPE=rabbitmq make mqtt
```

Comment:
Add empty line

Reply:
@WashingtonKK this is still not addressed